phone-book/src/app.js

```diff
@@ -0,0 +1,177 @@
+class App {
+  constructor(appContainerId) {
+    this.state = {};
+    this.appContainer = document.getElementById(appContainerId);
+    this.router = new Router(this);
+    this.serverAPI = new ServerAPI();
+    const userPage = new EditUser(this);
+    this.pages = {
+      contacts : {
+        pageObject : new Contacts(this),
+        href : '/contacts.html'
+      },
+      keypad : {
+        pageObject : new Keypad(this),
+        href : '/keypad.html'
+      },
+      editUser : {
+        pageObject : userPage,
+        href : '/edit-contact.html'
+      },
+      user : {
+        pageObject : new User(this),
+        href : '/user.html'
+      },
+      addUser : {
+        pageObject : userPage,
+        href : '/add-user.html'
+      }
+    }
+
+    const appHtml = this.renderAppContainer();
+    this.appContainer.innerHTML = appHtml;
+    this.addRouter();
+
+    this.state.pageName = 'Contacts';
+    this.updateState({activePage : 'contacts'});
```

Comment:
you don't have to update state in constructor you can set it directly in this.state property